code/development-env/docker/docker-compose-app.yml

```diff
@@ -1,49 +1,83 @@
-version: '3'
+version: '3.7'
+
+x-common-variables: &common
+  AUTHORISATION_SERVICE: http://datalab-auth:9000
+  AUTH_SIGNIN_URL: http://datalabs.localhost
+  DATABASE_HOST: mongodb
+  INFRASTRUCTURE_API: http://datalab-infra:8003
+  KUBERNETES_API: http://host.docker.internal:8001
+  LOG_LEVEL: ${LOG_LEVEL:-info}
+  VAULT_API: http://vault:8200
+  VAULT_APP_ROLE: ${VAULT_APP_ROLE}
+
 services:
   datalab-auth:
     image: node:8
+    working_dir: /usr/src/app
+    command: ["sh", "-c", "yarn start"]
     environment:
-      - AUTHORISATION_API_CLIENT_ID
-      - AUTHORISATION_API_CLIENT_SECRET
-      - AUTHORISATION_API_IDENTIFIER
-      - USER_MANAGEMENT_API_CLIENT_ID
-      - USER_MANAGEMENT_API_CLIENT_SECRET
-      - LOG_LEVEL
+      <<: *common
+      AUTHORISATION_API_CLIENT_ID: ${AUTHORISATION_API_CLIENT_ID}
+      AUTHORISATION_API_CLIENT_SECRET: ${AUTHORISATION_API_CLIENT_SECRET}
+      AUTHORISATION_API_IDENTIFIER: ${AUTHORISATION_API_IDENTIFIER}
+      USER_MANAGEMENT_API_CLIENT_ID: ${USER_MANAGEMENT_API_CLIENT_ID}
+      USER_MANAGEMENT_API_CLIENT_SECRET: ${USER_MANAGEMENT_API_CLIENT_SECRET}
     ports:
       - 9000:9000
-    entrypoint: sh -c
-    command:
-      - cd /usr/src/app && yarn start
     volumes:
-      - $PWD/../../auth-service:/usr/src/app/:ro
+      - ${PWD}/../../auth-service:/usr/src/app/:ro
+    networks:
+      - private
 
   datalab-infra:
     image: node:8
+    working_dir: /usr/src/app
+    command: ["sh", "-c", "yarn start"]
     environment:
-      - VAULT_APP_ROLE
-      - LOG_LEVEL
+      <<: *common
     ports:
       - 8003:8003
-    depends_on:
-      - datalab-auth
-    entrypoint: sh -c
-    command:
-      - cd /usr/src/app && yarn start
     volumes:
       - $PWD/../../infrastructure-api:/usr/src/app/:ro
+    networks:
+      - private
+    links:
+      - datalab-auth
+      - mongodb
+      - vault
 
-  datalab-app-api:
+  datalab-api:
     image: node:8
+    working_dir: /usr/src/app
+    command: ["sh", "-c", "yarn api"]
     environment:
-      - VAULT_APP_ROLE
-      - LOG_LEVEL
+      <<: *common
     ports:
-      - 3000:3000
       - 8000:8000
-    depends_on:
+    volumes:
+      - $PWD/../../datalab-app:/usr/src/app:ro
+    networks:
+      - private
+    links:
       - datalab-auth
-    entrypoint: sh -c
-    command:
-      - cd /usr/src/app && yarn start
+      - datalab-infra
+      - mongodb
+      - vault
+
+  datalab-app:
+    image: node:8
+    working_dir: /usr/src/app
+    command: ["sh", "-c", "yarn app"]
+    ports:
+      - 3000:3000
     volumes:
       - $PWD/../../datalab-app:/usr/src/app:ro
+      - $PWD/../config/local/web_auth_config.json:/usr/src/app/public/web_auth_config.json:ro
+    networks:
+      - private
+
+networks:
```

Comment:
I've now idea what this does, could you please add a comment explaining why a regular network isn't sufficient

Reply:
This is no longer needed, I have set all the container to run on the default network.